source/common/init/manager_impl.cc

```diff
@@ -51,18 +54,34 @@ void ManagerImpl::initialize(const Watcher& watcher) {
 
     // Attempt to initialize each target. If a target is unavailable, treat it as though it
     // completed immediately.
+    uint32_t target_index = 0;
     for (const auto& target_handle : target_handles_) {
       if (!target_handle->initialize(watcher_)) {
-        onTargetReady();
+        onTargetReady(target_names_[target_index]);
```

Comment:
If we expose the target_name from TargetHandle(the only implementation contains the name anyway), 
the `target_names_` is not needed then. WDTY @mergeconflict

Reply:
Agree, that will avoid duplicate memory usage.